### PR TITLE
drivers: spi: Test rx bigger than tx

### PR DESCRIPTION
**Pull request by dgastonochoa**
This is a proposal to add a test case for SPI.

This test cases checks that the SPI driver works properly when providing an RX buffer whose length is bigger than the TX one. As far as I understand, the expected behavior in this circumstance is that SPI sends all the data in TX and, after it has been consumed, it keeps sending zeroes so RX can be filled with incoming data.

Test plan
=========

Connect an STM32H7 board to the PC with pins D11 and D12 connected.

Open a minicom or equivalent terminal to read the test report.

Execute the following commands:

```
rm -rf ~/zephyrproject/zephyr/build
west build -p auto -b nucleo_h753zi tests/drivers/spi/spi_loopback
west flash

rm -rf ~/zephyrproject/zephyr/build
west build -p auto -b nucleo_h753zi -T tests/drivers/spi/spi_loopback/drivers.spi.stm32_spi_16bits_frames.loopback
west flash

rm -rf ~/zephyrproject/zephyr/build
west build -p auto -b nucleo_h753zi -T tests/drivers/spi/spi_loopback/drivers.spi.stm32_spi_dma.loopback
west flash
```

And verify all tests pass.

### CODE REVIEW

**Comment by dgastonochoa** [comment]
Kind reminder to review this @tbursztyka, @teburd. Thanks.

**Review by teburd — CHANGES_REQUESTED** [review]
Minor nits, otherwise the test looks fine to me

File: tests/drivers/spi/spi_loopback/src/spi.c [hunk 1]
```diff
@@ -428,6 +429,79 @@ static int spi_rx_every_4(struct spi_dt_spec *spec)
 	return 0;
 }
 
+static int spi_rx_bigger_than_tx(struct spi_dt_spec *spec)
+{
+	const uint32_t tx_buf_size = 8;
+
+	_Static_assert(tx_buf_size < BUF_SIZE);
```
Comment (teburd) [inline tests/drivers/spi/spi_loopback/src/spi.c:436]:
```suggestion
	BUILD_ASSERT(tx_buf_size < BUF_SIZE, "Transmit buffer is expected to be smaller than the receive buffer");
```

File: tests/drivers/spi/spi_loopback/src/spi.c [hunk 2]
```diff
@@ -428,6 +428,71 @@ static int spi_rx_every_4(struct spi_dt_spec *spec)
...
+			.buf = buffer_tx,
+			.len = 8,
+		},
+	};
+	const struct spi_buf rx_bufs[] = {
+		{
+			.buf = buffer_rx,
+			.len = BUF_SIZE,
+		}
+	};
```
[the remark this answers is not in the document]
Comment (dgastonochoa) [reply tests/drivers/spi/spi_loopback/src/spi.c:449]: Done

**Review by teburd — DISMISSED** [review]

File: tests/drivers/spi/spi_loopback/src/spi.c (hunk 1 above)
Comment (dgastonochoa) [reply tests/drivers/spi/spi_loopback/src/spi.c]: Didn't see the "commit suggestion" button so I applied the change manually.

**Review by tbursztyka — DISMISSED** [review]

**Comment by fabiobaltieri** [comment]
@dgastonochoa sorry this also needs a rebase

**Comment by dgastonochoa** [comment]
> @dgastonochoa sorry this also needs a rebase

Done

**Review by fabiobaltieri — APPROVED** [review]

**Comment by dgastonochoa** [comment]
Kind reminder to @teburd and @tbursztyka to review this. This was already approved but I needed to rebase.  

Thanks.

**Review by teburd — APPROVED** [review]

**Pull request merged**